### PR TITLE
Introduce passport token definition

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Review by andrewpatto — APPROVED** [review]

File: AAI/AAIConnectProfile.md [hunk 1]
```diff
@@ -696,22 +745,69 @@ If this information is a JWT, especially from the recommended Token Endpoint, th
...
+Passport Issuers MUST issue a Passport token conforming to the requirements in this section when a token exchange
+with the `requested_token_type=urn:ga4gh:params:oauth:token-type:passport` is successfully performed
+(as described in the [Conformance for Passport Issuers](#conformance-for-passport-issuers) section).
+
+###### Header
+
+This spec prescribes the following JWS headers for Passport tokens
+in addition to the guidelines established in [RFC7515](https://datatracker.ietf.org/doc/html/rfc7515):
+
+- `typ`: REQUIRED where the value must be `application/vnd.ga4gh.passport+jwt` for Passport tokens.
```
Comment (martin-kuba) [inline AAI/AAIConnectProfile.md:775]: The [RFC7515 section 4.1.9](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.9) recommends to omit the "application/" prefix of a media type value in a "typ" Header Parameter when no other '/' appears in the media type   value. So I think this should be just `vnd.ga4gh.passport+jwt`

Comment (martin-kuba) [inline AAI/AAIConnectProfile.md:775]:
```suggestion
- `typ`: REQUIRED where the value must be `vnd.ga4gh.passport+jwt` for Passport tokens.
```

File: AAI/AAIConnectProfile.md [hunk 2]
```diff
@@ -458,27 +465,27 @@ TODO embedded-access-token is this a passport access token or a visa access toke
...
 
-2.  Passport Issuers do not need to be a be a OIDC provider, and MAY provide a .well-known endpoint that doesn't conform to the OIDC Discovery specification for ease of finding signing keys.  
-
-    1. Passport Issuers MAY be AAI Clients, Clearinghouses, Brokers or any other entity and do not need to be part of the OIDC flow.  
-    
-    2. Passports SHOULD be signed in the same way that Brokers sign access tokens.
+2.  Passport Issuers SHOULD be Brokers.
 
-3.  Passports themselves are JWTs that contain Visas. Passports use this format [Format](#claims-sent-to-data-holder-by-a-broker-via-token-or-userinfo) as a signed JWT.  
+3.  Passports themselves are JWTs that contain Visas. Passports use [this format](#claims-sent-to-data-holder-by-a-broker-via-token-or-userinfo) as a signed JWT.
```
Comment (martin-kuba) [inline AAI/AAIConnectProfile.md:476]: missing `-endpoint` at the end of the link
```suggestion
3.  Passports themselves are JWTs that contain Visas. Passports use [this format](#claims-sent-to-data-holder-by-a-broker-via-token-or-userinfo-endpoint) as a signed JWT.
```

File: AAI/AAIConnectProfile.md [hunk 3]
```diff
@@ -669,18 +710,26 @@ Payload:
 
 -   `addtional claims`: OPTIONAL. Any other additional non-GA4GH claims are allowed. This specification does not dictate the format of other claims.
 
+<hr style="width: 10em; margin: 2em auto;"/>    
+
 #### Claims sent to Data Holder by a Broker via Token or UserInfo Endpoint
 
+##### Claims via UserInfo Endpoint
+
+The endpoint MAY use `application/json` or `application/jwt`. It is RECOMMENDED that if desiring to return a JWT, a
```
Comment (martin-kuba) [inline AAI/AAIConnectProfile.md:719]: This section most repeats what is already in [OIDC Core section 5.3.2 "Successful UserInfo Response"](https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse) so I think just the first paragraph is actually needed.

**Review by martin-kuba — CHANGES_REQUESTED** [review]
made a few suggested changes

File: jekyll-builder-local/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM jekyll/builder:4
+FROM jekyll/builder:4.2.0
```
Comment (andrewpatto) [inline jekyll-builder-local/Dockerfile:1]: The 4.2.2 jekyll builder (published two weeks ago) has some incompatibility with our setup - so pinning this to the latest 4.x that did work

File: AAI/AAIConnectProfile.md (hunk 3 above)
Comment (mbarkley) [reply AAI/AAIConnectProfile.md]: @martin-kuba I removed most of this section, but I kept the bullet point on `<ga4gh-spec-claims>` and made it more specific (so it refers to `ga4gh_passport_v1`). WDYT?

File: AAI/AAIConnectProfile.md [hunk 5]
```diff
@@ -462,23 +462,21 @@ TODO embedded-access-token is this a passport access token or a visa access toke
...
-    1. Passport Issuers MAY be AAI Clients, Clearinghouses, Brokers or any other entity and do not need to be part of the OIDC flow.  
-    
-    2. Passports SHOULD be signed in the same way that Brokers sign access tokens.
-
-3.  Passports themselves are JWTs that contain Visas. Passports use this format [Format](#claims-sent-to-data-holder-by-a-broker-via-token-or-userinfo) as a signed JWT.  
+3.  Passports themselves are JWTs that contain Visas. Passports use [this format](#claims-sent-to-data-holder-by-a-broker-via-token-or-userinfo-endpoint) as a signed JWT.
     
     1. It is RECOMMENDED for Passports to conform to the <https://tools.ietf.org/html/rfc7515> (JWS) Specification.  
+
+    2. Passports MUST be signed with the `RS256` or `ES256` algorithm.
```
Comment (TomConner) [inline AAI/AAIConnectProfile.md:471]: 384 or 512 should be allowed but not required. ES256 ought to be listed first.

File: AAI/AAIConnectProfile.md [hunk 6]
```diff
@@ -671,16 +697,33 @@ Payload:
...
 #### Claims sent to Data Holder by a Broker via Token or UserInfo Endpoint
 
-Only the GA4GH claims truly must be as prescribed here. Refer to OIDC Spec for
-more information. The endpoint MAY use `application/json` or
-`application/jwt`.  It is RECOMMENDED that if desiring to return a JWT, a Token Endpoint supporting AAI token exchange exists to do that and that the UserInfo Endpoint returns a general blob.
+##### Claims via UserInfo Endpoint
+
+The [UserInfo](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) endpoint MAY use `application/json`
+or `application/jwt`. It is RECOMMENDED that if desiring to return a JWT, a Token Endpoint supporting
+AAI token exchange exists to do that and that the UserInfo Endpoint returns an `application/json` blob.
```
Comment (TomConner) [inline AAI/AAIConnectProfile.md:704]: I suggest "response" rather than "blob".

File: AAI/AAIConnectProfile.md [hunk 7]
```diff
@@ -671,16 +697,33 @@ Payload:
...
+###### Header
+
+This spec prescribes the following JWS headers for Passport tokens
+in addition to the guidelines established in [RFC7515](https://datatracker.ietf.org/doc/html/rfc7515):
+
+- `typ`: REQUIRED where the value must be `vnd.ga4gh.passport+jwt` for Passport tokens.
+
+###### Claims
+
+Only the GA4GH claims truly must be as prescribed here. The
```
Comment (TomConner) [inline AAI/AAIConnectProfile.md:725]: I suggest striking "truly".

**Review by TomConner — CHANGES_REQUESTED** [review]

**Comment by TomConner** [comment]
This branch is published at https://ga4gh.github.io/data-security/1.2-draft-passport-token-definition/aai-openid-connect-profile.

File: AAI/AAIConnectProfile.md (hunk 5 above)
Comment (mbarkley) [reply AAI/AAIConnectProfile.md:471]: We're going to leave this language here and send a separate PR to replace the language referencing algorithms that is currently distributed throughout the document with a single section that specifies recommended algorithms and a reference to the RFC8725.

**Review by TomConner — APPROVED** [review]

**Pull request merged**